### PR TITLE
feat(common): LDML keyboard_processor infrastructure 🙀

### PR DESCRIPTION
**Pull request by mcdurdin**
Feature description: #7042.

Establishes baseline class and headers, and stub for loading LDML keyboards embedded in kmxplus files.

* Enables identification of KMXPlus files in processor factory
* Adds a stub action to generate an 'a' output character
* Adds a passing null_keyboard unit test for a hypothetical ldml keyboard
* Writes various template files for testing and producing installable test artifacts

@keymanapp-test-bot skip

### CODE REVIEW

File: core/src/ldml/ldml_processor.hpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/*
+  Copyright:    © 2018 SIL International.
+  Description:  Internal keyboard class and adaptor class for the API.
+  Create Date:  2 Oct 2018
+  Authors:      Tim Eves (TSE)
+  History:      2 Oct 2018 - TSE - Refactored out of km_kbp_keyboard_api.cpp
```
Comment (srl295) [inline core/src/ldml/ldml_processor.hpp:6]: copypasta

File: core/src/ldml/ldml_processor.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+#include <unordered_map>
+#include <keyman/keyboardprocessor.h>
+
+#include "processor.hpp"
+#include "option.hpp"
+
+namespace km {
+namespace kbp {
+
+#define KM_KBP_LDML_VERSION u"1.0"
```
Comment (srl295) [inline core/src/ldml/ldml_processor.hpp:22]: would be better KM_KBP_LMDL_PROCESSOR_VERSION or at least a comment to that effect

File: core/src/ldml/ldml_processor.cpp [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  file.seekg(0, std::ios::beg);
+
+  data.reserve((size_t)size);
+  if(!file.read((char *) data.data(), size)) {
+    return false;
+  }
+
+  file.close();
+
+  const kmx::PCOMP_KEYBOARD comp_keyboard = (kmx::PCOMP_KEYBOARD)data.data();
```
Comment (srl295) [inline core/src/ldml/ldml_processor.cpp:59]: <static_cast> ?

Comment (mcdurdin) [reply core/src/ldml/ldml_processor.cpp:59]: 😭

**Pull request merged**